src/adaptors/http2/http2_adaptor.c

```diff
@@ -2482,49 +2505,56 @@ static void qdr_http2_adaptor_final(void *adaptor_context)
  */
 static void qdr_http2_adaptor_init(qdr_core_t *core, void **adaptor_context)
 {
-    qdr_http2_adaptor_t *adaptor = NEW(qdr_http2_adaptor_t);
-    adaptor->core    = core;
-    adaptor->adaptor = qdr_protocol_adaptor(core,
-                                            "http2",                // name
-                                            adaptor,               // context
-                                            qdr_http_activate,
-                                            qdr_http_first_attach,
-                                            qdr_http_second_attach,
-                                            qdr_http_detach,
-                                            qdr_http_flow,
-                                            qdr_http_offer,
-                                            qdr_http_drained,
-                                            qdr_http_drain,
-                                            qdr_http_push,
-                                            qdr_http_deliver,
-                                            qdr_http_get_credit,
-                                            qdr_http_delivery_update,
-                                            qdr_http_conn_close,
-                                            qdr_http_conn_trace);
-    adaptor->log_source = qd_log_source(QD_HTTP_LOG_SOURCE);
-    adaptor->protocol_log_source = qd_log_source("PROTOCOL");
-    adaptor->lock = sys_mutex();
-    *adaptor_context = adaptor;
-    DEQ_INIT(adaptor->listeners);
-    DEQ_INIT(adaptor->connectors);
-    DEQ_INIT(adaptor->connections);
+    sys_mutex_lock(core->adaptor_startup_lock);
```

Comment:
@ganeshmurthy this is going to cause a deadlock right?
This PR is not causing that crash, but it is hanging during router initialization.

Reply:
Put the lock in the wrong place. Moved it up. Please test now. Thanks

Reply:
@ganeshmurthy worked great this time!